h2o-core/src/main/java/water/fvec/Frame.java

```diff
@@ -1633,19 +1635,28 @@ public int getCurrentRowSize() throws IOException {
       assert av > 0;
       return _line.length;
     }
+    
 
     byte[] getBytesForRow() {
       StringBuilder sb = new StringBuilder();
       BufferedString tmpStr = new BufferedString();
-      for (int i = 0; i < _curChks.length; i++ ) {
+      for (int i = 0; i < _curChks.length; i++) {
         Vec v = _curChks[i]._vec;
-        if(i > 0) sb.append(_parms._separator);
-        if(!_curChks[i].isNA(_chkRow)) {
-          if( v.isCategorical() ) sb.append('"').append(v.factor(_curChks[i].at8(_chkRow))).append('"');
-          else if( v.isUUID() ) sb.append(PrettyPrint.UUID(_curChks[i].at16l(_chkRow), _curChks[i].at16h(_chkRow)));
-          else if( v.isInt() ) sb.append(_curChks[i].at8(_chkRow));
-          else if (v.isString()) sb.append('"').append(_curChks[i].atStr(tmpStr, _chkRow)).append('"');
-          else {
+        if (i > 0) sb.append(_parms._separator);
+        if (!_curChks[i].isNA(_chkRow)) {
+          if (v.isCategorical()) {
+            final String escapedString = escapeQuotesForCsv(v.factor(_curChks[i].at8(_chkRow)));
+            sb.append('"')
```

Comment:
I know this si knitpiccking, but these would look fine on a single line (same below)

Reply:
My "fault", originally, this was formatted in a single line. Fixed.